### PR TITLE
SNOW-1258702:SnowSQL File Upload support for China

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-1258702

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/connector/s3_storage_client.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def __init__(
             )
         )
         self.use_s3_regional_url = use_s3_regional_url
+        self.location_type = stage_info["locationType"]
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/s3_storage_client.py:83]: is this `locationType` information guaranteed to be provided in the stage_info?
shall we use stage_info.get("locationType")?

File: src/snowflake/connector/s3_storage_client.py [hunk 2]
```diff
@@ -92,6 +93,9 @@ def __init__(
     def transfer_accelerate_config(
         self, use_accelerate_endpoint: bool | None = None
     ) -> bool:
+        # accelerate cannot be used in China and us government
+        if "S3China" in self.location_type:
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/s3_storage_client.py:97]: is this case sensitive, also just curious where do you find out this constant value

File: src/snowflake/connector/s3_storage_client.py (hunk 1 above)
Comment (sfc-gh-yuwang) [reply src/snowflake/connector/s3_storage_client.py:83]: As I talked with people from storage team, this is a stable way to decide where the endpoint is, will use get()

File: src/snowflake/connector/s3_storage_client.py (hunk 2 above)
Comment (sfc-gh-yuwang) [reply src/snowflake/connector/s3_storage_client.py:97]: Got this information from storage team as well

Comment (sfc-gh-aling) [inline src/snowflake/connector/s3_storage_client.py:97]: nit: stage_info.get returns None if locationType is not in the dict, so we shall check whether self.location_type has a value
```suggestion
        if self.location_type and "S3China" in self.location_type:
```

**Comment by sfc-gh-aling** [comment]
I'm good with the change, let's merge it after that addressing the nit

**Pull request merged**